.github/workflows/build-docker-image-pipeline.yml

```
@@ -0,0 +1,25 @@
+name: Build and Push Docker Image
+on:
+  push:
+    branches:
+      - main
+      - develop
+
+jobs:
+  build-docker-image-job:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout ✅
+        uses: actions/checkout@v3
+
+      - name: Login to GitHub Container Registry
+        run: echo "${{ secrets.GHCR_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+
+      - name: Build Image
+        run: |
+          docker build -t ghcr.io/fh-joanneum-iron-road-for-children/iron-road-for-children-web/web .
```

You should add a version tag here. At least to distinguish between main and develop e.g.
```suggestion
          docker build -t ghcr.io/fh-joanneum-iron-road-for-children/iron-road-for-children-web/web:$GITHUB_REF_NAME .
```

This would create an image tagged with `main` for pushes to main and when pushed to develop it will be tagged with `develop`.
Depends on your CD if this is enough or if you also need some version number e.g. `1.0.0` to destinguish between different builds. If that is needed, I would change the workflow to only run on tag pushes (the code above then would also work and tag the image with the pushed gittag name).

Of course you would also need to add this tag to the push command below

---

Using the branch name is a good idea. Also the backend teams implemented it with branch names